### PR TITLE
feat: Specify merge-method

### LINKED ISSUE #2: Feature request: Allow merge method to be customised
It would be really nice if we could allow the merge method to be customised, currently it uses the default method which is a merge commit.

The default is being used because we are not supplying a `mergeMethod` property.
image

https://docs.github.com/en/graphql/reference/input-objects#enablepullrequestautomergeinput

### PR DESCRIPTION
**Pull request by alexwilson**
# What?

Specify the merge-method when enabling auto-merge (resolves #2).

- Allow setting it via a `merge-method` input variable.
- When unset, attempt to read the default for the current repository.
- Fall-back to `merge` (current & default Github behaviour).
- Also refactor out the main body of this action into a class and make it friendlier to work on this offline.

# Why?

Workflows are unique and specific to codebases, and so it's best not to force `MERGE` as on everybody who uses this!

### CODE REVIEW

File: src/enable-github-automerge-action.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        await this.enableAutoMerge(pullRequestId, mergeMethod)
+        debug(
+          `Successfully enabled auto-merge for pull-request #${pullRequest.number}`
+        );
+  
+    }
+  
+    private async getMergeMethod(repo) {
+      
+      const preferredMergeMethod = getInput("merge-method", { required: false });
```
Comment (alexwilson) [inline src/enable-github-automerge-action.ts:50]: This call to `getInput` isn't ideal - and probably should have happened in `main.ts`

Comment (JakeChampion) [reply src/enable-github-automerge-action.ts:50]: Perhaps it could be an optional argument to the constructor?

Comment (JakeChampion) [reply src/enable-github-automerge-action.ts:50]: By that I mean, make `preferredMergeMethod` an instance property which can be set by an optional argument in the constructor and then in the main.ts file you pass the value as `getInput("merge-method", { required: false })`

Comment (alexwilson) [reply src/enable-github-automerge-action.ts:50]: It started it out that way, but it felt unnecessary...  Think I will add it back!  A few extra lines of code for a slight sounder sleep!

**Comment by alexwilson** [comment]
Small smoke-test-ish thing here: https://github.com/alexwilson/frontend/pull/1481

There _has_ to be a nicer way of developing for actions...

**Comment by JakeChampion** [comment]
> Small smoke-test-ish thing here: [alexwilson/frontend#1481](https://github.com/alexwilson/frontend/pull/1481)

Nice!

> There _has_ to be a nicer way of developing for actions...

Yeah, they are currently really annoying to develop and test properly

File: src/enable-github-automerge-action.ts (hunk 1 above)
Comment (alexwilson) [reply src/enable-github-automerge-action.ts]: Done!

**Comment by alexwilson** [comment]
Mind having another look at this @JakeChampion?

README has some further instructions

**Review by JakeChampion — APPROVED** [review]
This is nice!

**Pull request merged**